### PR TITLE
Drop OneCycleLR pickling of bound methods

### PR DESCRIPTION
**Pull request by MikhailKardash**
This PR drops pickling of bound methods in OneCycleLR scheduler. Adds a new attribute 'anneal_strategy' and re-inits whichever method is specified by this attribute. 

This fixes an issue where creating a subclass of OneCycleLR makes that subclass unpickleable.

Added tests for `state_dict()` and `load_state_dict()`

### CODE REVIEW

**Comment by albanD** [comment]
> This fixes an issue where creating a subclass of OneCycleLR makes that subclass unpickleable.

That is surprising to me. OneCycleLR is pickleable but not subclass?

**Comment by MikhailKardash** [comment]
Converting to draft while I better document the issue I'm trying to solve

**Comment by MikhailKardash** [comment]
I dove into the issue a bit deeper, and it turns out it's due to the pickler trying to pickle the optimizer when pickling state_dict().
Issue opened here:  https://github.com/pytorch/pytorch/issues/94450
